WordPress/gutenberg
# Implement pattern overrides behind `IS_GUTENBERG_PLUGIN` flag

## What?
See https://github.com/WordPress/gutenberg/issues/53705#issuecomment-1983576920.

Puts the pattern overrides feature behind the `IS_GUTENBERG_PLUGIN` flag to prevent it shipping to core.

## How?
I've taken the approach of using `IS_GUTENBERG_PLUGIN` where the `core/pattern-overrides` block bindings source is registered.

In other places in the codebase, we check for the existence of that `core/pattern-overrides` block binding source to disable the feature.

## Testing Instructions
This needs some thorough testing to ensure all traces of pattern overrides are no longer present.

To replicate `IS_GUTENBERG_PLUGIN` being `false` you can do the following:
- Edit the `webpack/shared.js` file, changing this line to `'process.env.IS_GUTENBERG_PLUGIN': false`. You can usually also modify this in the project package.json itself, but for me this caused a build failure.
https://github.com/WordPress/gutenberg/blob/ed7edbcc3db3155831239092830557df21a73b3a/tools/webpack/shared.js#L67-L68 
- Edit the `lib/load.php` file, changing this line to `define( 'IS_GUTENBERG_PLUGIN', false );`:
https://github.com/WordPress/gutenberg/blob/ed7edbcc3db3155831239092830557df21a73b3a/lib/load.php#L12

Then finally rebuild.
